### PR TITLE
test case for loading java 9 to 11 classes 

### PR DESCRIPTION
**Pull request by rakshitkr**
Test case for registering the virtual file system.
Test cases for loading particular class of different JDK version from CI

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -400,6 +400,41 @@
                 </configuration>
             </plugin>
         </plugins>
+        <pluginManagement>
```
Comment (anddann) [inline pom.xml]: It this change necessary?
What does it do?
Currently, we do not want any eclipse specific settings in our pom?

Comment (rakshitkr) [reply pom.xml]: Yes, you are right... I have committed them by mistake. Will remove it

Comment (anddann) [reply pom.xml]: thx...

**Comment by anddann** [comment]
Thx @rakshitkr  for your help!
I've adapted your test case slightly.
However, I don't have JDK 11 installed.
Can you please test the merged branch fix/newAsm on your local JDK 11 installation?
Would be super cool!
Thx!

**Comment by rakshitkr** [comment]
Sure.. Will test them right now

**Comment by rakshitkr** [comment]
@anddann I'm getting exception SootMethodRefImpl$ClassResolutionFailedException for testLoadingJava9ClassFromCI testcase. I don't get this exception when "-allow-phantom-refs" is used

**Comment by anddann** [comment]
Please provide the full stack trace

**Comment by rakshitkr** [comment]
[eclipseLogs1.txt](https://github.com/anddann/soot/files/2829009/eclipseLogs1.txt)

[Thread-10] ERROR heros.solver.CountingThreadPoolExecutor - Worker thread execution failed: Class java.lang.invoke.MethodHandle doesn't have method invoke([]) : java.lang.Object; failed to resolve in superclasses and interfaces
soot.SootMethodRefImpl$ClassResolutionFailedException: Class java.lang.invoke.MethodHandle doesn't have method invoke([]) : java.lang.Object; failed to resolve in superclasses and interfacesLooking in java.lang.invoke.MethodHandle which has methods [<java.lang.invoke.MethodHandle: java.lang.invoke.MethodType type()>, <java.lang.invoke.MethodHandle: void <init>(java.lang.invoke.MethodType,java.lang.invoke.LambdaForm)>, <java.lang.invoke.MethodHandle: java.lang.Object invokeExact(java.lang.Object[])>, <java.lang.invoke.MethodHandle: java.lang.Object invoke(java.lang.Object[])>, <java.lang.invoke.MethodHandle: java.lang.Object invokeBasic(java.lang.Object[])>, <java.lang.invoke.MethodHandle: java.lang.Object linkToVirtual(java.lang.Object[])>, <java.lang.invoke.MethodHandle: java.lang.Object linkToStatic(java.lang.Object[])>, <java.lang.invoke.MethodHandle: java.lang.Object linkToSpecial(java.lang.Object[])>, <java.lang.invoke.MethodHandle: java.lang.Object linkToInterface(java.lang.Object[])>, <java.lang.invoke.MethodHandle: java.lang.Object invokeWithArguments(java.lang.Object[])>, <java.lang.invoke.MethodHandle: java.lang.Object invokeWithArguments(java.util.List)>, <java.lang.invoke.MethodHandle: java.lang.invoke.MethodHandle asType(java.lang.invoke.MethodType)>, <java.lang.invoke.MethodHandle: java.lang.invoke.MethodHandle asTypeCached(java.lang.invoke.MethodType)>, <java.lang.invoke.MethodHandle: java.lang.invoke.MethodHandle asTypeUncached(java.lang.invoke.MethodType)>, <java.lang.invoke.MethodHandle: java.lang.invoke.MethodHandle asSpreader(java.lang.Class,int)>, <java.lang.invoke.MethodHandle: java.lang.invoke.MethodHandle asSpreader(int,java.lang.Class,int)>, <java.lang.invoke.MethodHandle: java.lang.invoke.MethodType asSpreaderChecks(java.lang.Class,int,int)>, <java.lang.invoke.MethodHandle: void spreadArrayChecks(java.lang.Class,int)>, <java.lang.invoke.MethodHandle: java.lang.invoke.MethodHandle withVarargs(boolean)>, <java.lang.invoke.MethodHandle: java.lang.invoke.MethodHandle asCollector(java.lang.Class,int)>, <java.lang.invoke.MethodHandle: java.lang.invoke.MethodHandle asCollector(int,java.lang.Class,int)>, <java.lang.invoke.MethodHandle: boolean asCollectorChecks(java.lang.Class,int,int)>, <java.lang.invoke.MethodHandle: java.lang.invoke.MethodHandle asVarargsCollector(java.lang.Class)>, <java.lang.invoke.MethodHandle: boolean isVarargsCollector()>, <java.lang.invoke.MethodHandle: java.lang.invoke.MethodHandle asFixedArity()>, <java.lang.invoke.MethodHandle: java.lang.invoke.MethodHandle bindTo(java.lang.Object)>, <java.lang.invoke.MethodHandle: java.lang.String toString()>, <java.lang.invoke.MethodHandle: java.lang.String standardString()>, <java.lang.invoke.MethodHandle: java.lang.String debugString()>, <java.lang.invoke.MethodHandle: java.lang.invoke.BoundMethodHandle bindArgumentL(int,java.lang.Object)>, <java.lang.invoke.MethodHandle: java.lang.invoke.MethodHandle setVarargs(java.lang.invoke.MemberName)>, <java.lang.invoke.MethodHandle: java.lang.invoke.MethodHandle viewAsType(java.lang.invoke.MethodType,boolean)>, <java.lang.invoke.MethodHandle: boolean viewAsTypeChecks(java.lang.invoke.MethodType,boolean)>, <java.lang.invoke.MethodHandle: java.lang.invoke.LambdaForm internalForm()>, <java.lang.invoke.MethodHandle: java.lang.invoke.MemberName internalMemberName()>, <java.lang.invoke.MethodHandle: java.lang.Class internalCallerClass()>, <java.lang.invoke.MethodHandle: java.lang.invoke.MethodHandleImpl$Intrinsic intrinsicName()>, <java.lang.invoke.MethodHandle: java.lang.invoke.MethodHandle withInternalMemberName(java.lang.invoke.MemberName,boolean)>, <java.lang.invoke.MethodHandle: boolean isInvokeSpecial()>, <java.lang.invoke.MethodHandle: java.lang.Object internalValues()>, <java.lang.invoke.MethodHandle: java.lang.Object internalProperties()>, <java.lang.invoke.MethodHandle: java.lang.invoke.MethodHandle copyWith(java.lang.invoke.MethodType,java.lang.invoke.LambdaForm)>, <java.lang.invoke.MethodHandle: java.lang.invoke.BoundMethodHandle rebind()>, <java.lang.invoke.MethodHandle: void updateForm(java.lang.invoke.LambdaForm)>, <java.lang.invoke.MethodHandle: void customize()>, <java.lang.invoke.MethodHandle: void <clinit>()>]
Looking in java.lang.Object which has methods [<java.lang.Object: void registerNatives()>, <java.lang.Object: void <init>()>, <java.lang.Object: java.lang.Class getClass()>, <java.lang.Object: int hashCode()>, <java.lang.Object: boolean equals(java.lang.Object)>, <java.lang.Object: java.lang.Object clone()>, <java.lang.Object: java.lang.String toString()>, <java.lang.Object: void notify()>, <java.lang.Object: void notifyAll()>, <java.lang.Object: void wait()>, <java.lang.Object: void wait(long)>, <java.lang.Object: void wait(long,int)>, <java.lang.Object: void finalize()>, <java.lang.Object: void <clinit>()>]

	at soot.SootMethodRefImpl.resolve(SootMethodRefImpl.java:238)
	at soot.SootMethodRefImpl.resolve(SootMethodRefImpl.java:152)
	at soot.jimple.internal.AbstractInvokeExpr.getMethod(AbstractInvokeExpr.java:56)
	at soot.jimple.validation.InvokeArgumentValidator.validate(InvokeArgumentValidator.java:54)
	at soot.jimple.JimpleBody.validate(JimpleBody.java:118)
	at soot.jimple.JimpleBody.validate(JimpleBody.java:98)
	at soot.baf.BafBody.<init>(BafBody.java:82)
	at soot.baf.Baf.newBody(Baf.java:549)
	at soot.PackManager.convertJimpleBodyToBaf(PackManager.java:1082)
	at soot.PackManager.runBodyPacks(PackManager.java:1037)
	at soot.PackManager.access$0(PackManager.java:910)
	at soot.PackManager$1.run(PackManager.java:664)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
Exception in thread "Thread-10" soot.SootMethodRefImpl$ClassResolutionFailedException: Class java.lang.invoke.MethodHandle doesn't have method invoke([]) : java.lang.Object; failed to resolve in superclasses and interfacesLooking in java.lang.invoke.MethodHandle which has methods [<java.lang.invoke.MethodHandle: java.lang.invoke.MethodType type()>, <java.lang.invoke.MethodHandle: void <init>(java.lang.invoke.MethodType,java.lang.invoke.LambdaForm)>, <java.lang.invoke.MethodHandle: java.lang.Object invokeExact(java.lang.Object[])>, <java.lang.invoke.MethodHandle: java.lang.Object invoke(java.lang.Object[])>, <java.lang.invoke.MethodHandle: java.lang.Object invokeBasic(java.lang.Object[])>, <java.lang.invoke.MethodHandle: java.lang.Object linkToVirtual(java.lang.Object[])>, <java.lang.invoke.MethodHandle: java.lang.Object linkToStatic(java.lang.Object[])>, <java.lang.invoke.MethodHandle: java.lang.Object linkToSpecial(java.lang.Object[])>, <java.lang.invoke.MethodHandle: java.lang.Object linkToInterface(java.lang.Object[])>, <java.lang.invoke.MethodHandle: java.lang.Object invokeWithArguments(java.lang.Object[])>, <java.lang.invoke.MethodHandle: java.lang.Object invokeWithArguments(java.util.List)>, <java.lang.invoke.MethodHandle: java.lang.invoke.MethodHandle asType(java.lang.invoke.MethodType)>, <java.lang.invoke.MethodHandle: java.lang.invoke.MethodHandle asTypeCached(java.lang.invoke.MethodType)>, <java.lang.invoke.MethodHandle: java.lang.invoke.MethodHandle asTypeUncached(java.lang.invoke.MethodType)>, <java.lang.invoke.MethodHandle: java.lang.invoke.MethodHandle asSpreader(java.lang.Class,int)>, <java.lang.invoke.MethodHandle: java.lang.invoke.MethodHandle asSpreader(int,java.lang.Class,int)>, <java.lang.invoke.MethodHandle: java.lang.invoke.MethodType asSpreaderChecks(java.lang.Class,int,int)>, <java.lang.invoke.MethodHandle: void spreadArrayChecks(java.lang.Class,int)>, <java.lang.invoke.MethodHandle: java.lang.invoke.MethodHandle withVarargs(boolean)>, <java.lang.invoke.MethodHandle: java.lang.invoke.MethodHandle asCollector(java.lang.Class,int)>, <java.lang.invoke.MethodHandle: java.lang.invoke.MethodHandle asCollector(int,java.lang.Class,int)>, <java.lang.invoke.MethodHandle: boolean asCollectorChecks(java.lang.Class,int,int)>, <java.lang.invoke.MethodHandle: java.lang.invoke.MethodHandle asVarargsCollector(java.lang.Class)>, <java.lang.invoke.MethodHandle: boolean isVarargsCollector()>, <java.lang.invoke.MethodHandle: java.lang.invoke.MethodHandle asFixedArity()>, <java.lang.invoke.MethodHandle: java.lang.invoke.MethodHandle bindTo(java.lang.Object)>, <java.lang.invoke.MethodHandle: java.lang.String toString()>, <java.lang.invoke.MethodHandle: java.lang.String standardString()>, <java.lang.invoke.MethodHandle: java.lang.String debugString()>, <java.lang.invoke.MethodHandle: java.lang.invoke.BoundMethodHandle bindArgumentL(int,java.lang.Object)>, <java.lang.invoke.MethodHandle: java.lang.invoke.MethodHandle setVarargs(java.lang.invoke.MemberName)>, <java.lang.invoke.MethodHandle: java.lang.invoke.MethodHandle viewAsType(java.lang.invoke.MethodType,boolean)>, <java.lang.invoke.MethodHandle: boolean viewAsTypeChecks(java.lang.invoke.MethodType,boolean)>, <java.lang.invoke.MethodHandle: java.lang.invoke.LambdaForm internalForm()>, <java.lang.invoke.MethodHandle: java.lang.invoke.MemberName internalMemberName()>, <java.lang.invoke.MethodHandle: java.lang.Class internalCallerClass()>, <java.lang.invoke.MethodHandle: java.lang.invoke.MethodHandleImpl$Intrinsic intrinsicName()>, <java.lang.invoke.MethodHandle: java.lang.invoke.MethodHandle withInternalMemberName(java.lang.invoke.MemberName,boolean)>, <java.lang.invoke.MethodHandle: boolean isInvokeSpecial()>, <java.lang.invoke.MethodHandle: java.lang.Object internalValues()>, <java.lang.invoke.MethodHandle: java.lang.Object internalProperties()>, <java.lang.invoke.MethodHandle: java.lang.invoke.MethodHandle copyWith(java.lang.invoke.MethodType,java.lang.invoke.LambdaForm)>, <java.lang.invoke.MethodHandle: java.lang.invoke.BoundMethodHandle rebind()>, <java.lang.invoke.MethodHandle: void updateForm(java.lang.invoke.LambdaForm)>, <java.lang.invoke.MethodHandle: void customize()>, <java.lang.invoke.MethodHandle: void <clinit>()>]
Looking in java.lang.Object which has methods [<java.lang.Object: void registerNatives()>, <java.lang.Object: void <init>()>, <java.lang.Object: java.lang.Class getClass()>, <java.lang.Object: int hashCode()>, <java.lang.Object: boolean equals(java.lang.Object)>, <java.lang.Object: java.lang.Object clone()>, <java.lang.Object: java.lang.String toString()>, <java.lang.Object: void notify()>, <java.lang.Object: void notifyAll()>, <java.lang.Object: void wait()>, <java.lang.Object: void wait(long)>, <java.lang.Object: void wait(long,int)>, <java.lang.Object: void finalize()>, <java.lang.Object: void <clinit>()>]

	at soot.SootMethodRefImpl.resolve(SootMethodRefImpl.java:238)
	at soot.SootMethodRefImpl.resolve(SootMethodRefImpl.java:152)
	at soot.jimple.internal.AbstractInvokeExpr.getMethod(AbstractInvokeExpr.java:56)
	at soot.jimple.validation.InvokeArgumentValidator.validate(InvokeArgumentValidator.java:54)
	at soot.jimple.JimpleBody.validate(JimpleBody.java:118)
	at soot.jimple.JimpleBody.validate(JimpleBody.java:98)
	at soot.baf.BafBody.<init>(BafBody.java:82)
	at soot.baf.Baf.newBody(Baf.java:549)
	at soot.PackManager.convertJimpleBodyToBaf(PackManager.java:1082)
	at soot.PackManager.runBodyPacks(PackManager.java:1037)
	at soot.PackManager.access$0(PackManager.java:910)
	at soot.PackManager$1.run(PackManager.java:664)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
soot.SootMethodRefImpl$ClassResolutionFailedException: Class java.lang.invoke.MethodHandle doesn't have method invoke([]) : java.lang.Object; failed to resolve in superclasses and interfacesLooking in java.lang.invoke.MethodHandle which has methods [<java.lang.invoke.MethodHandle: java.lang.invoke.MethodType type()>, <java.lang.invoke.MethodHandle: void <init>(java.lang.invoke.MethodType,java.lang.invoke.LambdaForm)>, <java.lang.invoke.MethodHandle: java.lang.Object invokeExact(java.lang.Object[])>, <java.lang.invoke.MethodHandle: java.lang.Object invoke(java.lang.Object[])>, <java.lang.invoke.MethodHandle: java.lang.Object invokeBasic(java.lang.Object[])>, <java.lang.invoke.MethodHandle: java.lang.Object linkToVirtual(java.lang.Object[])>, <java.lang.invoke.MethodHandle: java.lang.Object linkToStatic(java.lang.Object[])>, <java.lang.invoke.MethodHandle: java.lang.Object linkToSpecial(java.lang.Object[])>, <java.lang.invoke.MethodHandle: java.lang.Object linkToInterface(java.lang.Object[])>, <java.lang.invoke.MethodHandle: java.lang.Object invokeWithArguments(java.lang.Object[])>, <java.lang.invoke.MethodHandle: java.lang.Object invokeWithArguments(java.util.List)>, <java.lang.invoke.MethodHandle: java.lang.invoke.MethodHandle asType(java.lang.invoke.MethodType)>, <java.lang.invoke.MethodHandle: java.lang.invoke.MethodHandle asTypeCached(java.lang.invoke.MethodType)>, <java.lang.invoke.MethodHandle: java.lang.invoke.MethodHandle asTypeUncached(java.lang.invoke.MethodType)>, <java.lang.invoke.MethodHandle: java.lang.invoke.MethodHandle asSpreader(java.lang.Class,int)>, <java.lang.invoke.MethodHandle: java.lang.invoke.MethodHandle asSpreader(int,java.lang.Class,int)>, <java.lang.invoke.MethodHandle: java.lang.invoke.MethodType asSpreaderChecks(java.lang.Class,int,int)>, <java.lang.invoke.MethodHandle: void spreadArrayChecks(java.lang.Class,int)>, <java.lang.invoke.MethodHandle: java.lang.invoke.MethodHandle withVarargs(boolean)>, <java.lang.invoke.MethodHandle: java.lang.invoke.MethodHandle asCollector(java.lang.Class,int)>, <java.lang.invoke.MethodHandle: java.lang.invoke.MethodHandle asCollector(int,java.lang.Class,int)>, <java.lang.invoke.MethodHandle: boolean asCollectorChecks(java.lang.Class,int,int)>, <java.lang.invoke.MethodHandle: java.lang.invoke.MethodHandle asVarargsCollector(java.lang.Class)>, <java.lang.invoke.MethodHandle: boolean isVarargsCollector()>, <java.lang.invoke.MethodHandle: java.lang.invoke.MethodHandle asFixedArity()>, <java.lang.invoke.MethodHandle: java.lang.invoke.MethodHandle bindTo(java.lang.Object)>, <java.lang.invoke.MethodHandle: java.lang.String toString()>, <java.lang.invoke.MethodHandle: java.lang.String standardString()>, <java.lang.invoke.MethodHandle: java.lang.String debugString()>, <java.lang.invoke.MethodHandle: java.lang.invoke.BoundMethodHandle bindArgumentL(int,java.lang.Object)>, <java.lang.invoke.MethodHandle: java.lang.invoke.MethodHandle setVarargs(java.lang.invoke.MemberName)>, <java.lang.invoke.MethodHandle: java.lang.invoke.MethodHandle viewAsType(java.lang.invoke.MethodType,boolean)>, <java.lang.invoke.MethodHandle: boolean viewAsTypeChecks(java.lang.invoke.MethodType,boolean)>, <java.lang.invoke.MethodHandle: java.lang.invoke.LambdaForm internalForm()>, <java.lang.invoke.MethodHandle: java.lang.invoke.MemberName internalMemberName()>, <java.lang.invoke.MethodHandle: java.lang.Class internalCallerClass()>, <java.lang.invoke.MethodHandle: java.lang.invoke.MethodHandleImpl$Intrinsic intrinsicName()>, <java.lang.invoke.MethodHandle: java.lang.invoke.MethodHandle withInternalMemberName(java.lang.invoke.MemberName,boolean)>, <java.lang.invoke.MethodHandle: boolean isInvokeSpecial()>, <java.lang.invoke.MethodHandle: java.lang.Object internalValues()>, <java.lang.invoke.MethodHandle: java.lang.Object internalProperties()>, <java.lang.invoke.MethodHandle: java.lang.invoke.MethodHandle copyWith(java.lang.invoke.MethodType,java.lang.invoke.LambdaForm)>, <java.lang.invoke.MethodHandle: java.lang.invoke.BoundMethodHandle rebind()>, <java.lang.invoke.MethodHandle: void updateForm(java.lang.invoke.LambdaForm)>, <java.lang.invoke.MethodHandle: void customize()>, <java.lang.invoke.MethodHandle: void <clinit>()>]
Looking in java.lang.Object which has methods [<java.lang.Object: void registerNatives()>, <java.lang.Object: void <init>()>, <java.lang.Object: java.lang.Class getClass()>, <java.lang.Object: int hashCode()>, <java.lang.Object: boolean equals(java.lang.Object)>, <java.lang.Object: java.lang.Object clone()>, <java.lang.Object: java.lang.String toString()>, <java.lang.Object: void notify()>, <java.lang.Object: void notifyAll()>, <java.lang.Object: void wait()>, <java.lang.Object: void wait(long)>, <java.lang.Object: void wait(long,int)>, <java.lang.Object: void finalize()>, <java.lang.Object: void <clinit>()>]

	at soot.SootMethodRefImpl.resolve(SootMethodRefImpl.java:238)
	at soot.SootMethodRefImpl.resolve(SootMethodRefImpl.java:152)
	at soot.jimple.internal.AbstractInvokeExpr.getMethod(AbstractInvokeExpr.java:56)
	at soot.jimple.validation.InvokeArgumentValidator.validate(InvokeArgumentValidator.java:54)
	at soot.jimple.JimpleBody.validate(JimpleBody.java:118)
	at soot.jimple.JimpleBody.validate(JimpleBody.java:98)
	at soot.baf.BafBody.<init>(BafBody.java:82)
	at soot.baf.Baf.newBody(Baf.java:549)
	at soot.PackManager.convertJimpleBodyToBaf(PackManager.java:1082)
	at soot.PackManager.runBodyPacks(PackManager.java:1037)
	at soot.PackManager.access$0(PackManager.java:910)
	at soot.PackManager$1.run(PackManager.java:664)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)

**Comment by anddann** [comment]
on JDK11 or JDK9 and which test fails (which class)?

**Comment by rakshitkr** [comment]
on JDK 9 (As expected)
`testLoadingJava9ClassFromCI` => `SUCCESS`
`testLoadingJava11ClassFromCI` => `FAILS`
`testLoadingJava9to11Class` => `SUCCESS` for `VarHandle` & `FAILS` for `ConstantBootstraps`

on JDK 11 (Not as expected)
`testLoadingJava9ClassFromCI` => `SUCCESS`
`testLoadingJava11ClassFromCI` => `EXCEPTION`
`testLoadingJava9to11Class` => `SUCCESS` for both

**Comment by anddann** [comment]
Thx for the detailed report!

**Comment by anddann** [comment]
The results for JDK 9 are expected, right?

However

`testLoadingJava11ClassFromCI => EXCEPTION` should work

**Comment by rakshitkr** [comment]
@anddann yes

**Comment by anddann** [comment]
Can you compile soot on JDK11?

**Pull request merged**